app/assets/javascripts/application.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+document.onreadystatechange = function () {
+  if (document.readyState === 'interactive') {
+    var editDocumentForm = document.querySelector('.edit_document')
+    if (editDocumentForm == null) { return } // only run on edit document page
+    var urlPreview = document.getElementById('url-preview-id')
+    var basePath = document.getElementById('base-path-id')
+    var documentTitle = document.getElementById('document-title-id')
+    var noTitle = document.getElementById('no-title-id')
+    var errorGeneratingPath = document.getElementById('error-generating-path-id')
+    var showErrorMessage = function () {
+      urlPreview.setAttribute('class', 'app-hidden')
+      noTitle.setAttribute('class', 'app-hidden')
+      errorGeneratingPath.removeAttribute('class')
+    }
+    documentTitle.onblur = function () {
+      var url = new URL(document.location.origin + editDocumentForm.getAttribute('data-generate-path-path'))
+      if (!documentTitle.value) {
+        noTitle.removeAttribute('class')
+        urlPreview.setAttribute('class', 'app-hidden')
+        errorGeneratingPath.setAttribute('class', 'app-hidden')
+        return
+      }
+      url.searchParams.append('title', documentTitle.value)
+      window.fetch(url, {
+        credentials: 'include',
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8'
+        }
+      }).then(function (response) {
+        if (!response.ok) {
```

Comment:
This should be redundant as we're using .then.

Reply:
So I had a closer look at this, and it's only used when the controller throws an unexpected error and returns an HTML response instead of valid JSON. If we want to handle unexpected errors, then it would be shorter to just put a '.catch' on the end of response.json().

Thanks for prompting me to look at this - I had no idea you can do this. On reflection, though, the great thing about promises is the ability to separate happy/sad code, so on balance I'd prefer to avoid throwing errors and use '.catch' as it keeps that logical separation.

Reply:
We discussed this and we'll leave it unchanged because the non-200 case doesn't hit the catch() call unless an error is raised explicitly. 

Seems like this confusion may be because non-200 would go to catch() when implemented like this in jQuery.